### ISSUE TITLE
Paginate issues

### ISSUE THREAD

**Issue by GautierDele** [babenkoivan/elastic-scout-driver-plus#17]
Hello,

Firstly, thanks for this awesome package that saves me a lot of time and do things properly 👍.
I know my config is not the same as you tested with but shouldn't be related.

| Software                   | Version
| --------------------| ---------------
| PHP                           | 7.3.21
| Elasticsearch            | 7.9.2
| Laravel                      | 8.3.0
| Laravel Scout           | 8.3.1
| Elastic Scout Driver | 1.2.0

**Describe the bug**
Using paginate and REST Api gives me empty objet as data.
The data is correctly fetched server side.

**To Reproduce**
```php
$search = Product::matchAllSearch();
return response($search->paginate($request->input('itemsPerPage')));
```

**Current behavior**
As you mentionned : `Unlike the standard Scout pagination, Elastic Scout Driver Plus paginates matches and not the models.`  
This might be the problem when giving it to the client.
One more thing, i don't think it's possible to load a relation which is mandatory using REST because of the lazy part.

**Expected behavior**
Getting the models with loaded relations.

The response:
```json
{"current_page":1,"data":[{},{},{},{},{}],"first_page_url":"http:\/\/localhost\/api\/datatable\/products?page=1","from":1,"last_page":220,"last_page_url":"http:\/\/localhost\/api\/datatable\/products?page=220","links":[{"url":null,"label":"Previous","active":false},{"url":"http:\/\/localhost\/api\/datatable\/products?page=1","label":1,"active":true},{"url":"http:\/\/localhost\/api\/datatable\/products?page=2","label":2,"active":false},{"url":"http:\/\/localhost\/api\/datatable\/products?page=3","label":3,"active":false},{"url":"http:\/\/localhost\/api\/datatable\/products?page=4","label":4,"active":false},{"url":"http:\/\/localhost\/api\/datatable\/products?page=5","label":5,"active":false},{"url":"http:\/\/localhost\/api\/datatable\/products?page=6","label":6,"active":false},{"url":"http:\/\/localhost\/api\/datatable\/products?page=7","label":7,"active":false},{"url":"http:\/\/localhost\/api\/datatable\/products?page=8","label":8,"active":false},{"url":"http:\/\/localhost\/api\/datatable\/products?page=9","label":9,"active":false},{"url":"http:\/\/localhost\/api\/datatable\/products?page=10","label":10,"active":false},{"url":null,"label":"...","active":false},{"url":"http:\/\/localhost\/api\/datatable\/products?page=219","label":219,"active":false},{"url":"http:\/\/localhost\/api\/datatable\/products?page=220","label":220,"active":false},{"url":"http:\/\/localhost\/api\/datatable\/products?page=2","label":"Next","active":false}],"next_page_url":"http:\/\/localhost\/api\/datatable\/products?page=2","path":"http:\/\/localhost\/api\/datatable\/products","per_page":5,"prev_page_url":null,"to":5,"total":1100}
```

Thanks,
Gautier

**Comment by babenkoivan** [comment]
Hi @GautierDele, thanks for reporting. It's definitely looks like a bug. I'll have a look.

Please note, that `data` will have not only the models, but also documents, highlights, etc., i.e. everything, that a match includes.

**Comment by GautierDele** [comment]
Hi @babenkoivan , shouldn't it be an option to render all the elasticsearch data ?
It will for sure make the response pretty huge and I don't think all the data is needed

Am i right about the eager loading of the relations ? Not possible right now ?

**Comment by babenkoivan** [comment]
@GautierDele, I need more time to think it over, but I'll try to deliver some solution within the next 2-3 days.

As for the eager loading, can you please create a separate feature request? I'll work on it after this one.
It's possible with some workarounds, but there is no an easy way to do that yet.

**Comment by babenkoivan** [comment]
Hi @GautierDele, the issue is now resolved in [v1.10.1](https://github.com/babenkoivan/elastic-scout-driver-plus/releases/tag/v1.10.1). As I stated before, `data` includes all matches fields. I realize, that most of the times developers do not need all the fields, but this is a default behaviour. If the paginator has matches as items, then it's logical to have full match representations in `data` as well. 

Nevertheless, it's up to a developer to format response if needed. You can create a transformer class or use a middleware to format the pagination response, or you can just do it in the controller if you wish. Something like this should work:

```php
$paginator = $search->paginate($request->input('itemsPerPage'));

$content = tap($paginator->toArray(), function (array &$content) {
    // include only models in the response
    $content['data'] = Arr::pluck($content['data'], 'model');
});

$response = response($content);
```

**Comment by GautierDele** [comment]
Thanks @babenkoivan !
I'll give it a try 😊
